pilot/cmd/istioctl/gendeployment/cmd.go

```diff
@@ -0,0 +1,184 @@
+	cmd.PersistentFlags().StringVar(&install.PilotTag, "pilot-tag", install.PilotTag, "The tag to use to pull the `pilot-discovery` container")
+	cmd.PersistentFlags().StringVar(&install.CaTag, "ca-tag", install.CaTag, "The tag to use to pull the `ca` container")
+	cmd.PersistentFlags().StringVar(&install.ProxyTag, "proxy-tag", install.ProxyTag, "The tag to use to pull the `proxy` container")
+	cmd.PersistentFlags().BoolVar(&install.Debug, "debug", install.Debug, "If true, uses debug images instead of release images")
+	cmd.PersistentFlags().Uint16Var(&install.NodePort, "ingress-node-port", install.NodePort,
+		"If provided, Istio ingress proxies will run as a NodePort service mapped to the port provided by this flag. "+
+			"Note that this flag is ignored unless the \"ingress\" feature flag is provided too.")
+	cmd.PersistentFlags().StringVarP(&out, "out", "o", "helm", `Output format. Acceptable values are:
+					"helm": produces contents of values.yaml
+					"yaml": produces Kubernetes deployments`)
+
+	// TODO: figure out how we want to package up the charts with the binary to make this easy
+	cmd.PersistentFlags().StringVar(&helmChartLocation, "helm-chart-dir", ".",
+		"The directory to find the helm charts used to render Istio deployments. -o yaml uses these to render the helm chart locally.")
+
+	_ = cmd.PersistentFlags().MarkHidden("hub")
+	_ = cmd.PersistentFlags().MarkHidden("mixer-tag")
+	_ = cmd.PersistentFlags().MarkHidden("pilot-tag")
+	_ = cmd.PersistentFlags().MarkHidden("ca-tag")
+	_ = cmd.PersistentFlags().MarkHidden("proxy-tag")
+	return cmd
+}
+
+func getValues(path string, i *installation) (string, error) {
+	if path == "" {
+		return valuesFromInstallation(i), nil
+	}
+
+	out, err := ioutil.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return string(out), nil
+}
+
+type installation struct {
+	Mixer       bool
+	Pilot       bool
+	CA          bool
+	Ingress     bool
+	Initializer bool
+
+	Namespace string
+	Debug     bool
+	NodePort  uint16
+
+	// todo: support hub per component
+	Hub      string // hub to pull images from
+	MixerTag string
+	PilotTag string
+	CaTag    string
+	ProxyTag string
+}
+
+func defaultInstall() *installation {
+	return &installation{
+		Mixer:       true,
+		Pilot:       true,
+		CA:          true,
+		Ingress:     true,
+		Initializer: false,
+
+		Namespace: "istio-system",
+		Debug:     false,
+		NodePort:  0,
+
+		Hub:      "gcr.io/istio-testing",
+		MixerTag: defaultTag,
+		PilotTag: defaultTag,
+		CaTag:    defaultTag,
+		ProxyTag: defaultTag,
+	}
+}
+
+func (i *installation) setFeatures(features []string) error {
+	if len(features) == 0 {
+		return nil
+	}
+
+	i.Mixer = false
+	i.Pilot = false
+	i.CA = false
+	i.Ingress = false
+	i.Initializer = false
+	for _, f := range features {
+		switch strings.ToLower(f) {
+		case "telemetry", "policy":
+			i.Mixer = true
+			i.Pilot = true
```

Comment:
you do plan on going more fine grained than thsi right? disablePolicyChecks = true, enableTracing=true, etc..

Reply:
Not in this pass, since it requires helm template changes first. But yes eventually we'll support that too.